Improve mean and integral documentations

fmahebert opened the pull request:
## Proposed changes

- Improve clarity and fix rendering in `mean_value` documentation
- Improve clarity in `definite_integral` documentation

Fixes #1662.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

kidder commented on src/NumericalAlgorithms/LinearOperators/DefiniteIntegral.hpp:
```diff
@@ -16,15 +16,22 @@ class Mesh;
- * \param mesh the Mesh of the manifold on which `integrand` is located.
- * \returns the definite integral of `integrand` on the manifold.
+ * Given a function \f$f\f$, compute its integral \f$I\f$ with respect to
+ * the reference coordinates \f$(\xi, \eta, \zeta)\f$. In 1 dimension,
```
Multiple times:  we use `logical coordinates` instead of `reference coordinates` throughout the code...

kidder commented on src/NumericalAlgorithms/LinearOperators/MeanValue.hpp:
```diff
@@ -22,17 +22,18 @@ class Mesh;
- * \returns the mean value of `f` on the manifold
- * \param f the grid function of which to find the mean.
- * \param mesh the Mesh of the manifold on which f is located.
+ * The mean w.r.t. a different set of coordinates \f$x\f$ can be computed by
```
This is not true.  You need to modify the integral in the divisor as well which cannot be done using this function...

fmahebert replied:
I agree. What do you think of the following:

- edit this text to reflect that this function only works in logical coords.
- add a new function (probably in a new PR) `mean_value(data, mesh, jacobian)` that does the divisor appropriately for the coordinates associated with the jacobian.

fmahebert committed:
Clarify coordinates used in (in)definitely_integral

fmahebert committed:
Clarify coordinates and fix rendering in mean_value

fmahebert commented:
I've made some changes in response to @kidder's comments, and some other improvements to notation and clarity. I've also added a note to the `indefinite_integral` function.

Because the fixup was just as big as the original commit, I squashed and force-pushed.

wthrowe commented:
The documentation fixes look good.

I wonder if `mean_value` is really worth having.  It only works in logical coordinates, and in that case all it does is divide the integral by `2^d`.  You generally don't want a generalized form that calculates the denominator for you because then you are recalculating the volume for each quantity you integrate, and most uses of the function call it multiple times.

kidder commented:
Yeah, we should probably remove `mean_value` for the reasons @wthrowe gives.  But I would do that in a separate PR or we create an issue to debate the merits of `mean_value`
